sys/include/memarray.h

```diff
@@ -110,6 +122,44 @@ static inline void memarray_free(memarray_t *mem, void *ptr)
     mem->free_data = ptr;
 }
 
+/**
+ * @brief Extend the memarray with a new memory region
+ *
+ * This function extends the memarray pool with a new memory region. The region
+ * must be able to fit the supplied number of elements of the size used when
+ * initializing this memarray.
+ *
+ * @pre `mem != NULL`
+ * @pre `data != NULL`
+ * @pre `num != 0`
+ *
+ * @param[in,out] mem   memarray pool to extend
+ * @param[in]     data  pointer to user-allocated data
+ * @param[in]     num   number of elements in @p data
+ */
+void memarray_extend(memarray_t *mem, void *data, size_t num);
+
+/**
+ * @brief Reduce the memarray space, subtracting the memory pool
+ *
+ * It is up to the user to free all chunks in the reduced pool. The function
+ * will check if all elements in the pool are freed.
+ *
+ * @param[in,out] mem   memarray pool to reduce
+ * @param[in]     data  pointer to the user-allocated data to reduce
+ * @param[in]     num   number of elements to reduce the data pool with
+ */
+int memarray_reduce(memarray_t *mem, void *data, size_t num);
+
+/**
+ * @brief Returns the number of blocks available
+ *
+ * @param[in]   mem     memarray pool
+ *
+ * @returns             Number of elements available in the memarray pool
+ */
+size_t memarray_available(memarray_t *mem);
```

Comment:
This one is nice on itself, I would split into a different commit.

Reply:
Done!